modules/indexing/src/test/java/org/apache/ignite/internal/processors/query/h2/H2JavaObjectSerializerTest.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2019 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.processors.query.h2;
+
+import org.junit.Test;
+import static org.junit.Assert.*;
+import org.apache.ignite.IgniteLogger;
+import org.apache.ignite.internal.util.typedef.internal.U;
+import org.mockito.Mockito;
+import static org.mockito.Mockito.*;
+import org.apache.ignite.testframework.junits.common.GridCommonAbstractTest;
+
+public class H2JavaObjectSerializerTest extends GridCommonAbstractTest {
+   
```

Comment:
need to test whether the log has the exception lines:
           "Failed to serialize BinaryObject with typeId: " + ((BinaryObject) obj).typeId() :
                "Failed to Serialize object: " + obj.getClass().getName();
                
                String errorMsg = "Failed to deserialize data: " + Arrays.toString(bytes);

Reply:
Added tests to ensure the log captures the expected exception messages. Please advise if further adjustments are needed.